lib/src/util.dart

```diff
@@ -23,18 +23,35 @@ import 'constants.dart';
 import 'file_query.dart';
 import 'patch.dart';
 
+/// Collector whose instance is passed to every source file to aggregate patches
+/// from the entire code modification.
+class PatchCollector {
+  List<Patch> skippedPatches = [];
+}
+
 /// Returns the result of applying all of the [patches]
 /// (insertions/deletions/replacements) to the contents of [sourceFile].
 ///
-/// Throws an [Exception] if any two of the given [patches] overlap.
-String applyPatches(SourceFile sourceFile, Iterable<Patch> patches) {
+/// Throws an [Exception] if any two of the given [patches] overlap and the
+/// `skipOverlaps` argument is false or not provided. If `skipOverlaps` is true,
+/// overlapping patches will be added to the [PatchCollector] so that
+/// after the code modification completes.
+String applyPatches(SourceFile sourceFile, Iterable<Patch> patches,
+    PatchCollector patchCollector,
```

Comment:
This util is publicly exported as the main means for testing suggestors, so adding a required param here would be a breaking change. Additionally, this change turns this util from a simple input->output function to something that also produces a side-effect.

We always call `applyPatchesAndSave` in the runInteractive loop, so maybe an alternative approach would be to prompt the user about the overlapping patch instead of immediately throwing and giving them the option to either A) skip and continue applying the rest of the patches, or B) quit immediately. Then we could still collect those skipped patches (just locally inside the interactive loop) and log them out at the end so that the user still ends up with a list of skipped patches that they can apply manually. That would probably be a better experience than what we do now and wouldn't break any existing contract we have. Thoughts on that?